src/uu/tsort/src/tsort.rs

```diff
@@ -112,34 +100,49 @@ pub fn uu_app() -> Command {
         )
 }
 
+/// Find the element `x` in `vec` and remove it, returning its index.
+fn remove<T>(vec: &mut Vec<T>, x: T) -> Option<usize>
+where
+    T: PartialEq,
+{
+    for i in 0..vec.len() {
+        if vec[i] == x {
+            vec.remove(i);
+            return Some(i);
+        }
+    }
+    None
```

Comment:
what about something like ?
```suggestion
        vec.iter().position(|item| *item == x).map(|i| {
        vec.remove(i);
        i
    })
```